identity-actor/src/actor/actor_builder.rs

```diff
@@ -0,0 +1,274 @@
+  }
+
+  /// Build the actor with a custom transport.
+  pub async fn build_with_transport<TRA>(self, transport: TRA) -> Result<Actor>
+  where
+    TRA: Transport + Sized + Clone + Send + Sync + 'static,
+    TRA::Output: AsyncRead + AsyncWrite + Unpin + Send + 'static,
+    TRA::Dial: Send + 'static,
+    TRA::Listener: Send + 'static,
+    TRA::ListenerUpgrade: Send + 'static,
+    TRA::Error: Send + Sync,
+  {
+    let (noise_keypair, peer_id) = {
+      let keypair = self.keypair.unwrap_or_else(Keypair::generate_ed25519);
+      let noise_keypair = NoiseKeypair::<X25519Spec>::new().into_authentic(&keypair).unwrap();
+      let peer_id = keypair.public().to_peer_id();
+      (noise_keypair, peer_id)
+    };
+
+    let executor = Box::new(|fut| {
+      tokio::spawn(fut);
+    });
+
+    let mut swarm: Swarm<RequestResponse<ActorRequestResponseCodec>> = {
+      let mut config: RequestResponseConfig = RequestResponseConfig::default();
+      config.set_request_timeout(self.config.timeout);
+
+      let behaviour = RequestResponse::new(
+        ActorRequestResponseCodec(),
+        iter::once((ActorProtocol(), ProtocolSupport::Full)),
+        config,
+      );
+
+      let transport = transport
+        .upgrade(upgrade::Version::V1)
+        .authenticate(NoiseConfig::xx(noise_keypair).into_authenticated())
+        .multiplex(YamuxConfig::default())
+        .boxed();
+
+      SwarmBuilder::new(transport, behaviour, peer_id)
+        .executor(executor.clone())
+        .build()
+    };
+
+    for addr in self.listening_addresses {
+      swarm.listen_on(addr).map_err(|err| Error::TransportError {
+        context: "unable to start listening",
+        source: err,
+      })?;
+    }
+
+    let (cmd_sender, cmd_receiver) = mpsc::channel(10);
+
+    let event_loop = EventLoop::new(swarm, cmd_receiver);
+    let swarm_commander = NetCommander::new(cmd_sender);
+
+    let actor = Actor::from_builder(
+      swarm_commander.clone(),
+      self.handler_map,
+      self.object_map,
+      peer_id,
+      self.config,
+    )
+    .await?;
+
+    let actor_clone = actor.clone();
+
+    let event_handler = move |event: InboundRequest| {
+      let actor = actor_clone.clone();
+
+      if event.request_mode == RequestMode::Asynchronous {
+        actor.handle_request::<AsynchronousInvocationStrategy>(event);
+      } else {
+        actor.handle_request::<SynchronousInvocationStrategy>(event);
+      };
+    };
+
+    executor.exec(event_loop.run(event_handler).boxed());
+
+    Ok(actor)
+  }
+}
+
+impl Default for ActorBuilder {
+  fn default() -> Self {
+    Self::new()
+  }
+}
+
+/// Used to attach handlers and hooks to an [`ActorBuilder`].
+pub struct HandlerBuilder<'builder, MOD: SyncMode, OBJ>
+where
+  OBJ: Clone + Send + Sync + 'static,
+  MOD: 'static,
+{
+  pub(crate) object_id: ObjectId,
+  pub(crate) handler_map: &'builder mut HandlerMap,
+  _marker_obj: PhantomData<&'static OBJ>,
+  _marker_mod: PhantomData<&'static MOD>,
```

Comment:
> Why are you using `PhantomData<&'static T>` here instead of just `PhantomData<T>`?

The `PhantomData` documentation says:

> ## Ownership and the drop check
> Adding a field of type PhantomData<T> indicates that your type owns data of type T. This in turn implies that when your type is dropped, it may drop one or more instances of the type T. This has bearing on the Rust compiler's [drop check] analysis.
>
> If your struct does not in fact own the data of type T, it is better to use a reference type, like PhantomData<&'a T> (ideally) or PhantomData<*const T> (if no lifetime applies), so as not to indicate ownership.

I just followed that suggestion. Not sure if there's an actual difference between `*const T` and `&'static T` in this case.